template/src/app/App.tsx

```diff
@@ -4,21 +4,18 @@ import {NavigationContainer} from '@react-navigation/native';
 import {createMaterialTopTabNavigator} from '@react-navigation/material-top-tabs';
 import {SafeAreaProvider} from 'react-native-safe-area-context';
 import {Home, Details} from 'app/pages';
-import {SafeAreaView} from 'react-native';
 
 const Tab = createMaterialTopTabNavigator();
 
 const App = (): JSX.Element => {
   return (
     <SafeAreaProvider>
-      <SafeAreaView style={{flex: 1}}>
```

Comment:
Interesting! This does match the navigation example more closely: https://reactnavigation.org/docs/handling-safe-area

So this is more for me to be more confident I know how this works: if we wrap it in SafeAreaView right here we are basically constraining the entire app to sit inside the insets, which is very...opinionated and maybe wrong. Screen components should be in SafeAreaView if and only if they want it ?

Reply:
So the problem is that I don't think materialTopTabNavigator is actually using SafeAreaView... I think we're facing this issue here: https://stackoverflow.com/questions/55694161/safeareaview-not-working-with-materialtoptabnavigator-at-root

With SafeAreaView wrapping TopTabBar:

![image](https://user-images.githubusercontent.com/55203625/140982230-ee45e516-42b8-4a27-b35e-8f18b15b53a5.png)

Without SafeAreaView wrapping TopTabBar (wrapping the screens have no effect):

![image](https://user-images.githubusercontent.com/55203625/140982688-8747df9d-fee8-49e4-a0d9-92c5dc38ca49.png)

Screenshots taken from my iPhone.

This is nothing like the example provided by you. I'm still investigating this. Not sure if I should use another example in the mean time.

Reply:
Interesting - I have switched to using all bottom navigators, but I just looked over my last work project and I did `useInsets` and some "fancy" stuff in order to get things right. Here's the nav (bottom tab), a custom status bar, and how it wrapped together:

```typescript

const Tab = createBottomTabNavigator();
const TabNavigator = () => {
  const insets = useSafeAreaInsets();

  return (
    <Tab.Navigator
      initialRouteName="Inicio"
      screenOptions={({route}) => ({
        headerShown: false,
        tabBarStyle: {
          borderTopWidth: 1,
          paddingTop: 0,
          height: 55 + insets.bottom,
        },
        tabBarLabelStyle: {
          marginTop: 0,
          paddingTop: 0,
          paddingBottom: 10 + insets.bottom,
        },
        tabBarItemStyle: {
          height: 55 + insets.bottom,
        },
      })}>
      <Tab.Screen name="Inicio" component={HomeScreen} />
      <Tab.Screen name="Filtros" component={FilterScreen} />
      <Tab.Screen name="OfertaStack" component={OfferStackScreen} />
      <Tab.Screen name="Estadisticas" component={StatisticsTabNavigator} />
    </Tab.Navigator>
  );
};

const CustomStatusBar = () => {
  const insets = useSafeAreaInsets();
  return (
    <View style={{backgroundColor: colors.blue.color, height: insets.top}}>
      <StatusBar
        backgroundColor={colors.blue.color}
        animated={true}
        barStyle="light-content"
      />
    </View>
  );
};

const App = () => {
  let [bootSplashIsVisible, setBootSplashIsVisible] = useState(true);
  let [bootSplashLogoIsLoaded, setBootSplashLogoIsLoaded] = useState(false);
  let opacity = useRef(new Animated.Value(1));
  let translateY = useRef(new Animated.Value(0));

  let navigatorLoaded = async () => {
    await BootSplash.hide();

    Animated.stagger(250, [
      Animated.spring(translateY.current, {
        useNativeDriver: true,
        toValue: -50,
      }),
      Animated.spring(translateY.current, {
        useNativeDriver: true,
        toValue: Dimensions.get('window').height,
      }),
    ]).start();

    Animated.timing(opacity.current, {
      useNativeDriver: true,
      toValue: 0,
      duration: 150,
      delay: 350,
    }).start(() => {
      setBootSplashIsVisible(false);
    });
  };

  useEffect(() => {
    bootSplashLogoIsLoaded;
  }, [bootSplashLogoIsLoaded]);

  return (
    <SafeAreaProvider>
      <CustomStatusBar />
      <NavigationContainer onReady={() => navigatorLoaded()}>
        <TabNavigator />
      </NavigationContainer>
    </SafeAreaProvider>
  );
};
```

I definitely remember playing around with the status bar insets and that being critical to things layout out correctly on all devices

Reply:
note the tab bar has bottom insets in use as well, might be needed - but top - for top tabs

Reply:
So should we implement something similiar? 🤔 What do you suggest?

Reply:
*if* we are demonstrating navigation (and though it is not minimal, I still think it is a positive - *everyone* needs navigation), then I think it is best to do it in a best practice style and I think doing a StatusBar that has the right insets is pretty important

I checked react-navigation just now  - I was really curious how they did it, 

https://github.com/react-navigation/react-navigation/blob/82a16690973a7935939a25a66d5786955b6c8ba7/example/src/index.tsx#L239

they use statusbar but don't use insets :thinking: 
react-native-paper provider, which is in use in react-navigation example does not put insets anywhere in it's stuff (I looked through PaperProvider and PortalHost, which is the whole view stack there in Paper Provider.

Strange!

Trial and error I guess. But direct to your question: Including a StatusBar seems like something all of these examples are doing, and seems like something necessary in every single project so it is okay in a boilerplate template even. It might fix the problem

Reply:
Thanks for pointing out. The template is using a StatusBar but in the Home page. I'll move it to App.tsx.

 I investigated this a little bit more and it seems that it's working right for IOS Simulator:

![image](https://user-images.githubusercontent.com/55203625/141002380-b93d454f-b655-4e8f-8399-747963c87da3.png)

But I still have this issue when I run it on device:
![image](https://user-images.githubusercontent.com/55203625/141002736-cee077bb-bc9c-4ddb-b251-02b4bfc017af.png)

I'll take a closer look tomorrow. 

Thanks, Mike!

Reply:
Aaand it's because I have dark mode on. 😂

Reply:
Ah, well, at least that's a really productive addition - I mean, everyone wants that now and there are at least good examples of it (the react-native template itself is a pretty good example perhaps - I know it supports it). That seems subtle though, I would not have thought of that! Always learning something :nerd_face:

Reply:
I'm still not happy with the changes I came with, because that restricts us to black or white on IOS if we want to fill that gap. Ultimately, I think useInsets should be used instead of wrapping TabNavigator in SafeAreaView, just like the example you provided.